src/display.rs

```diff
@@ -522,14 +524,18 @@ where
     I::Item: Borrow<u8>,
 {
     let mut encoder = BufEncoder::<N>::new();
-    encoder.put_bytes(bytes, case);
-    let encoded = encoder.as_str();
-
-    if let Some(precision) = f.precision() {
-        if encoded.len() > precision {
-            return f.pad_integral(true, "0x", &encoded[..precision]);
+    let bytes = bytes.into_iter();
```

Comment:
This is not wrong but no longer needed since you can just call `.into_iter().take(n)` for the limited case. You may change it if you want to but it's not required.